src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -61,63 +60,90 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
         {
             ExpressionSyntax expressionSyntax = TargetExpressionSyntax;
             VariableDeclarationSyntax declarationSyntax = _syntax.Declaration;
+            bool hasAwait = _syntax.AwaitKeyword.Kind() != default;
 
             Debug.Assert((expressionSyntax == null) ^ (declarationSyntax == null)); // Can't have both or neither.
 
             bool hasErrors = false;
-            BoundMultipleLocalDeclarations declarationsOpt = null;
-            BoundExpression expressionOpt = null;
-            Conversion iDisposableConversion = Conversion.NoConversion;
-            TypeSymbol iDisposable = this.Compilation.GetSpecialType(SpecialType.System_IDisposable); // no need for diagnostics, so use the Compilation version
+            TypeSymbol iDisposable;
+            if (hasAwait)
+            {
+                iDisposable = this.Compilation.GetWellKnownType(WellKnownType.System_IAsyncDisposable);
+                hasErrors |= ReportUseSiteDiagnostics(iDisposable, diagnostics, _syntax.AwaitKeyword.GetLocation());
```

Comment:
>hasErrors |= ReportUseSiteDiagnostics(iDisposable, diagnostics, _syntax.AwaitKeyword.GetLocation()); [](start = 16, length = 100)

Is there a reason to differ from regular IDisposable in regard to reporting use-site diagnostics?

Reply:
The comment made me think it was unnecessary. But it looks like there was a bug for IDisposable case. 

---
In reply to: [160562446](https://github.com/dotnet/roslyn/pull/23961#discussion_r160562446) [](ancestors = 160562446)